python/triton/compiler/code_generator.py

```diff
@@ -1205,7 +1205,10 @@ def ast_to_ttir(fn, signature, specialization, constants, debug, target):
     new_attrs = {k: [("tt.divisibility", 16)] for k in specialization.divisible_by_16}
     for k in specialization.divisible_by_8:
         attr = new_attrs[k] if k in new_attrs else []
-        attr.append(("tt.max_divisibility", 8))
+        if k in specialization.divisible_by_16:
```

Comment:
I suspect this is a workaround. Add a TODO here?

Reply:
Sure